### DISCUSSION TITLE
what time the 5.0 release?

### DISCUSSION THREAD

**Discussion by chenbimo** [gulpjs/gulp#2606] [Help] [answered]
somebody know?

**Comment by phated** [comment, marked as answer]
When it is done.

**Reply by ZhangChengLin** [reply]
Where to learn 5.0

**Reply by phated** [reply]
It's not done, but it's not going to be very different than 4.0

**Reply by jimmywarting** [reply]
What is going to be new about v5? haven't heard anything about a v5
(haven't use gulp in a very long time)

**Reply by phated** [reply]
@jimmywarting Core team members that were working on parts needed for the new release had life stuff come up. It'll happen when it happens.

**Reply by jimmywarting** [reply]
wasn't asking when it will be released, i where merely wondering about what is so new about v5 that it cause for a major release.
i don't see any changelog, v5-beta branch or any issue like it

**Reply by jimmywarting** [reply]
nvm, i found what i was looking for: https://github.com/orgs/gulpjs/projects/4

**Reply by phated** [reply]
Oh, I completely misread the question. Sorry.

New:
Switching from readable-stream to streamx
Dropping support for anything less than Node 10 so we can have some goodies like recursive mkdir
Landing some major bumps that have been outstanding

Cleanup:
Switching entire project (60+ repos) over to prettier, GitHub Actions, and a few more cleanup things that happens as part of our scaffold upgrade.

> i don't see any changelog, v5-beta branch or any issue like it

The project is spread over 60+ repos in the gulpjs org, plus another 20-40 I maintain on external orgs. Work happens directly in each repository. All of our public roadmaps exist at https://github.com/orgs/gulpjs/projects

**Reply by jimmywarting** [reply]
>Oh, I completely misread the question. Sorry.

No worries. with so many packages scatter around then it was unclear where one should look. But I found projects at last.

**Reply by begin-again** [reply]
@phated Node 10 is EOL and Node 12 goes EOL in April. Shouldn't you target Node 14+?

**Reply by phated** [reply]
@begin-again No, gulp doesn't follow node's EOL flow because it is too fast and plenty of people get left behind. Gulp 5 is a transition release to drop some really old support we have. We'll do a v6 upgrade that has even bigger changes.

**Reply by ZhangChengLin** [reply]
> @begin-again No, gulp doesn't follow node's EOL flow because it is too fast and plenty of people get left behind. Gulp 5 is a transition release to drop some really old support we have. We'll do a v6 upgrade that has even bigger changes.

Is gulp updated by updating the dependent components?
Is the version number of gulp the same, but the version of the component it actually depends on is inconsistent?
If so, why not mark it in a prominent position?

**Reply by phated** [reply]
I don't understand your questions, but I'll try to answer what I think you are asking. Gulp is a thin shim over the gulp CLI and some other libraries (like glob-watcher, vinyl-fs, and undertaker), the underlying libraries have their semver managed strictly and get bumped major when breaking changes occur. Since gulp is a thin wrapper over these libraries, any breaking changes are breaking changes to gulp, so we batch all the majors from underlying libraries into one major gulp release.

**Reply by TokugawaTakeshi** [reply]
@phated 

How much dollars you need to finish v5.0?
If this sum is available for non-reach people, maybe chip in will be possible.

**Reply by phated** [reply]
@TokugawaTakeshi I've been doing a lot of open source consulting recently, and I'd charge a client USD$60,000 or more to complete the remaining work.

**Reply by TokugawaTakeshi** [reply]
@phated Maybe I misunderstood, so please let me to ask by other words. Do you need to be invested 6 thousands dollars to finish the Gulp 5.X?

**Reply by phated** [reply]
Open source doesn't have timelines, it is not a job. If you want a timeline, that is something you would have to pay for. You asked what it would cost and I gave you a cost estimate that I'd give customers—it would cost *60 thousand dollars. Otherwise, you can wait patiently or start completing work in our project board.

**Comment by yocontra** [comment]
I was planning on helping out more to get v5 across the finish line but had a change of jobs + just found out I have a kid on the way - sorry! As Blaine said above, the release won't include any "major" new features or changes, the major bump is mostly because we are dropping support for older node versions.

**Comment by Alpine418** [comment]
Are there any plans to support ESM import/export natively. More and more libs do not support CommonJS anymore.
